WP4/D4.6/report.tex

```diff
@@ -0,0 +1,186 @@
+Similarly for generated output, conflicts can be automatically resolved without the consequences of automatic resolution in the source code.
+In the worst case scenario, a notebook can be re-executed to produce new output.
+
+
+\begin{figure}
+    \center
+    \includegraphics[width=10cm]{img/nbmerge-web}
+    \caption{nbmerge web-based merge conflict resolution tool allows individual conflict resolution.}
+    \label{fig:nbmerge-web}
+\end{figure}
+
+When merge conflicts do occur, the nbmerge web tool can be used to view and resolve the conflicting changes one by one.
+
+% section merging_notebooks (end)
+
+
+\section{version control integration} % (fold)
+\label{sub:version_control_integration}
+
+nbdime provides integration with the git version control system for its diffing and merging tools.
+Git provides two mechanisms for integrating external software for
+performing the diff and merge operations on specific files.
+
+The first is called a "driver", and is used internally by git when computing diffs and performing merges. nbdime can register its command-line diff rendering as a diff driver for notebook files.
+This can be done as a system-wide configuration, or for individual git repositories.
+nbdime also provides a merge driver for auto-resolving conflicts on notebooks,
+ensuring that a \texttt{git merge} never results in an invalid notebook document,
+and that conflicts can be resolved in the user's familiar notebook-editing environment.
+
+The second git integration point is called a "tool", e.g. "difftool" or "mergetool".
+These are like drivers, but allow external applications that may open windows to be launched.
+nbdime provides tool entrypoints for diff and merge to launch the web interfaces for viewing diffs
+and manually resolving merge conflicts.
+
+% subsection version_control_integration (end)
+
+
+\section{Future work}
+The current version of nbdime is a working prototype but we expect
+some more work to be necessary to make it a polished product.
+Designers from the wider Jupyter community will be involved in
+polishing the visual presentation of the web interfaces,
+and users will become involved in further testing once the tool
+reaches a wider user base. Furthermore we expect to be able to
+improve the algorithms used for diffing and merging further,
+both for improving the diff quality and performance.
+
+While nbdime integrates well into the local git workflow of diffing and merging,
+much of the diffing and merging work of today's scientists and developers happens
+on public websites such as GitHub or Bitbucket,
+where the user cannot customize.
+However, these websites do support custom diff viewing of certain file types,
+such as images and GeoJSON files \cite{geojson-diff}.
+After some maturation of the design of the nbdime-web diff viewer,
+we will work with GitHub and others to integrate diff viewing of notebooks into their websites,
+as we have done in the past with rendering of single notebooks.
+
```

Comment:
Add something like?: By integrating nbdime with remote notebook editing tools like JupyterLab, comparing notebooks on remote servers should become much easier than today, and would integrate well with such a development process.